src/app/shell/personal-cabinet/provider/create-provider/create-provider.component.ts

```diff
@@ -151,8 +151,15 @@ export class CreateProviderComponent extends CreateFormComponent implements OnIn
   checkEmpty(form: FormGroup) { 
     let res = [];     
     for (let el in form?.controls) {
-      if (!form.get(el).value && form.get(el).status !== 'VALID') {
-        res.push(el)
+      if ((!form.get(el).value 
+        && form.get(el).status !== 'VALID')
+        || (form?.get(el).value.street === ''
+        || form?.get(el).value.buildingNumber === ''
+        || form?.get(el).value.city === ''
+        || form?.get(el).value.district === ''
+        || form?.get(el).value.region === '')
+      ) {
+        res.push(el);
```

Comment:
you can simplify it, use for example find
and go via controls to fins the first unvalid